src/jit/lower.cpp

```diff
@@ -1990,6 +1990,7 @@ GenTree* Lowering::LowerTailCallViaHelper(GenTreeCall* call, GenTree* callTarget
 //
 // Notes:
 //    - Decomposes long comparisons that feed a GT_JTRUE (32 bit specific).
+//    - Decomposes long comparisons that produce a value (X86 specific).
```

Comment:
Is there a particular reason that we decompose these operators here rather than in the decomposition proper?